examples/rp/src/bin/pio_ws2812.rs

```diff
@@ -77,7 +79,9 @@ impl<'d, P: Instance, const S: usize> Ws2812<'d, P, S> {
     pub async fn write(&mut self, colors: &[RGB8]) {
         for color in colors {
             let word = (u32::from(color.g) << 24) | (u32::from(color.r) << 16) | (u32::from(color.b) << 8);
-            self.sm.tx().wait_push(word).await;
+            if !self.sm.tx().try_push(word) {
+                self.sm.tx().wait_push(word).await;
+            }
```

Comment:
Unfortunately it seems they are. I think if the fifo is ever empty it will stay low long enough to reset (actually more like hold). The reset time is only 50us.

Reply:
Well, that's only true if nothing else is happening on the MCU. If you have a bunch of other tasks running you might not get back to the fifo for who knows how long.

Reply:
Can you remove it? it is very non-idiomatic, having this in example code can confuse users into thinking it is required for some reason.

If you're concerned about things breaking without `--release`, we can add a comment.